app/controllers/concerns/sufia/files_controller_behavior.rb

```diff
@@ -59,6 +61,9 @@ def load_resource_from_solr
 
     # routed to /files/new
     def new
+      find_collections_with_edit_access
```

Comment:
Uses Hydra::Collections to get the list of collections the current user is allowed to update.